failure in thread_mutex/test_async.modify_event

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8463/label=hpc-test-node,worker=3/consoleFull
```
18:30:50 [ RUN      ] thread_mutex/test_async.modify_event/0
18:30:51 /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node/worker/3/contrib/../test/gtest/ucs/test_async.cc:559: Failure
18:30:51 Expected: (le.count()) > (count), actual: 2 vs 2
18:30:52 
[  FAILED  ] thread_mutex/test_async.modify_event/0, where GetParam() = 2 (2161 ms)
18:30:52 [ RUN      ] thread_mutex/test_async.two_timers/0
...
make: *** [test] Error 1
18:44:44 make: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node/worker/3/build-test/test/gtest'
```

--- a/test/gtest/ucs/test_async.cc
+++ b/test/gtest/ucs/test_async.cc
@@ -561,7 +561,13 @@ UCS_TEST_P(test_async, modify_event) {
     ucs_async_modify_handler(le.event_id(), POLLIN);
     count = le.count();
     le.push_event();
-    suspend_and_poll(&le, COUNT);
+    for (int i = 0; i < TIMER_RETRIES; ++i) {
+        suspend_and_poll(&le, 1);
+        if (le.count() > count) {
+            break;
+        }
+        UCS_TEST_MESSAGE << "retry " << (i + 1);
+    }
     EXPECT_GT(le.count(), count);
 
     ucs_async_modify_handler(le.event_id(), 0);

